src/classes/OPP_PrimaryContactRoleMerge.cls

```diff
@@ -0,0 +1,256 @@
+                if (ocr.Role == ocrRole) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /*******************************************************************************************************
+        * @description Gets the OCRs with role equals blank.
+        * @return List<OpportunityContactRole> The OCRs with role role = blank.
+        ********************************************************************************************************/
+        public List<OpportunityContactRole> getBlankRole() {
+            List<OpportunityContactRole> result = new List<OpportunityContactRole>();
+            for (OpportunityContactRole ocr : record.OpportunityContactRoles) {
+                if (String.isBlank(ocr.Role)) {
+                    result.add(ocr);
+                }
+            }
+            return result;
+        }
+
+        /*******************************************************************************************************
+        * @description Gets the OCRs with Role equals to the Role in the parameter.
+        * @param String OCR Role.
+        * @return List<OpportunityContactRole> The OCRs with Role equals to the Role in the parameter.
+        ********************************************************************************************************/
+        public List<OpportunityContactRole> getRoleEquals(String ocrRole) {
+            List<OpportunityContactRole> result = new List<OpportunityContactRole>();
+            for (OpportunityContactRole ocr : record.OpportunityContactRoles) {
+                if (ocr.Role == ocrRole) {
+                    result.add(ocr);
+                }
+            }
+            return result;
+        }
+
+        /*******************************************************************************************************
+        * @description Gets the OCRs with Role not equals to the Role in the parameter.
+        * @param String OCR Role.
+        * @return List<OpportunityContactRole> The OCRs with Role not equals to the Role in the parameter.
+        ********************************************************************************************************/
+        public List<OpportunityContactRole> getRoleNotEquals(String ocrRole) {
+            List<OpportunityContactRole> result = new List<OpportunityContactRole>();
+            for (OpportunityContactRole ocr : record.OpportunityContactRoles) {
+                if (ocr.Role != ocrRole) {
+                    result.add(ocr);
+                }
+            }
+            return result;
+        }
+    }
+
+
+    /*******************************************************************************************************
+    * @description Error Log. This class contains the logic that creates the error logs.
+    ********************************************************************************************************/
+    public class Errors {
```

Comment:
Could we just collect a list of errors and process all of the dml at the same time?